docs/messaging.rst

```diff
@@ -1052,9 +1064,26 @@ Message type: ``status``::
         execution_state : ('busy', 'idle', 'starting')
     }
 
+When a kernel receives a request and begins processing it,
+it shall immediately publish a status message with ``execution_state: 'busy'``.
+When that kernel has completed processing the request
+and has finished publishing associated IOPub messages, if any,
+it shall publish a status message with ``execution_state: 'idle'``.
+Thus, the outputs associated with a given execution shall generally arive
+between the idle and busy status messages associated with a given request.
+
+.. note::
+
+    There is a caveat for asynchronous output,
```

Comment:
This file is the spec, so I don't think there's a link to add, since it would be to here.